typings/index.d.ts

```diff
@@ -3,16 +3,75 @@
 
 import * as React from 'react';
 
+// Static hoisting
+interface REACT_STATICS {
+  childContextTypes: true;
+  contextType: true;
+  contextTypes: true;
+  defaultProps: true;
+  displayName: true;
+  getDefaultProps: true;
+  getDerivedStateFromError: true;
+  getDerivedStateFromProps: true;
+  mixins: true;
+  propTypes: true;
+  type: true;
+}
+
+interface KNOWN_STATICS {
+  name: true;
+  length: true;
+  prototype: true;
+  caller: true;
+  callee: true;
+  arguments: true;
+  arity: true;
+}
+
+interface MEMO_STATICS {
+  $$typeof: true;
+  compare: true;
+  defaultProps: true;
+  displayName: true;
+  propTypes: true;
+  type: true;
+}
+
+interface FORWARD_REF_STATICS {
+  $$typeof: true;
+  render: true;
+  defaultProps: true;
+  displayName: true;
+  propTypes: true;
+}
+
+type NonReactStatics<
+  S extends React.ComponentType<any>,
+  C extends {
+    [key: string]: true;
+  } = {}
+> = {
+  [key in Exclude<
```

Comment:
Could you export the types in conditions to make the code more readable?

Reply:
Check the new commit.